### ISSUE TITLE
FloatCanvas Ellipse objects not being drawn

### ISSUE THREAD

**Issue by scaramallion** [wxWidgets/Phoenix#614]
FloatCanvas ellipses aren't being drawn correctly. The following should (and used to) draw an ellipse overlapping with the circle and bound within the rectangle. However now the ellipse isn't drawn at all.
```python
import wx
from wx.lib.floatcanvas import FloatCanvas as FC

class Main(wx.Frame):
    def __init__(self, parent, title):
        wx.Frame.__init__(self, parent, -1, title, size=(400, 400))
        canvas = FC.FloatCanvas(self, size=(400, 400))
        obj = FC.Circle([0, 0], 50)
        canvas.AddObject(obj)
        obj = FC.Ellipse([0, 0], [50, 75])
        canvas.AddObject(obj)
        obj = FC.Rectangle([0, 0], [50, 75])
        canvas.AddObject(obj)
        canvas.Draw(force=True)

if __name__ == "__main__":
    app = wx.App()
    main = Main(None, 'FC Test')
    main.Centre()
    main.Show()
    app.MainLoop()
````
Changing to a positive width and negative height is the only way to draw an ellipse, i.e. `obj = FC.Ellipse([0, 0], [50, -75])`. 

It looks like `wx.DC.DrawEllipse` may have changed to no longer allow negative width/heights which are produced by the `Ellipse._Draw` method.

Ubuntu 16.04
wxPython 4.0.0rc1.dev3477+0aa026a gtk3 (phoenix) from pypi
Python 3.5.2

**Comment by RobinD42** [comment]
Yeah, it works correctly on the other platforms, so my guess would be that it is GTK3 specific. The GTK version of `DoDrawEllipse` has code that compensates for negative values, and it is common code for both GTK2 and GTK3, but perhaps there is something in the supporting code that needs to be done differently for GTK3.  

Please file a ticket for this at trac.wxwidgets.org and post a link here for tracking. If you can give them a simple sample using plain DC API calls (no FloatCanvas) it will help.
